src/nile/common.py

```diff
@@ -28,25 +28,31 @@
     # subject to change
     "0x041a78e741e5af2fec34b695679bc6891742439f7afb8484ecd7766661ad02bf"
 )
+DEFAULT_GATEWAYS = {
+    "goerli2": "https://alpha4-2.starknet.io",
+    "integration": "https://external.integration.starknet.io",
+}
 
 
 def get_gateways():
     """Get the StarkNet node details."""
     try:
         with open(NODE_FILENAME, "r") as f:
             gateway = json.load(f)
-            return gateway
-
+            gateways = {**gateway, **DEFAULT_GATEWAYS}
```

Comment:
If there's a conflicting network, the `DEFAULT_GATEWAYS` network would overwrite it in the local variable `gateways` when the gateways dict is passed to the querying function. Thus, `"goerli2": "new_url"` in node.json will return the goerli2 from `DEFAULT_NETWORKS`

I'll switch the order so that the node.json's gateway overwrites the `DEFAULT_GATEWAYS` network of the same name